File: app/src/main/java/faith/changliu/androiddemo/fragments/FragB.java
package faith.changliu.androiddemo.fragments;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.design.widget.TextInputEditText;
import android.support.design.widget.TextInputLayout;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.webkit.WebChromeClient;
import android.webkit.WebView;
import android.webkit.WebViewClient;
import android.widget.Button;
import android.widget.ProgressBar;

import faith.changliu.androiddemo.R;
import faith.changliu.androiddemo.helpers.CommonUtils;
import faith.changliu.androiddemo.helpers.DownloadUtils;

public class FragB extends Fragment implements View.OnClickListener {

    private WebView mWebView;
    private ProgressBar mProgressBar;
    private Button mBtnLoadUrl;
    private TextInputLayout mEtUrlInputLayout;
    private TextInputEditText mEtUrlInput;


    public static FragB newInstance() {
        return new FragB();
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.frag_b, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        mBtnLoadUrl = view.findViewById(R.id.mBtnLoadUrl);
        mEtUrlInputLayout = view.findViewById(R.id.mEtUrlInputLayout);
        mEtUrlInput = view.findViewById(R.id.mEtUrlInput);
        mProgressBar = view.findViewById(R.id.progressBar);
        mWebView = view.findViewById(R.id.webView);

        mWebView.getSettings().setJavaScriptEnabled(true);
        mWebView.setWebChromeClient(new WebChromeClient() {
            @Override
            public void onProgressChanged(WebView view, int newProgress) {
                super.onProgressChanged(view, newProgress);
                if (newProgress > 0 && newProgress < 100) {
                    mProgressBar.setProgress(newProgress);
                    mProgressBar.setVisibility(View.VISIBLE);
                } else {
                    mProgressBar.setVisibility(View.GONE);
                }
            }
        });

        mWebView.setWebViewClient(new WebViewClient());

        mBtnLoadUrl.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        if (v.getId() == R.id.mBtnLoadUrl) {
            mEtUrlInputLayout.clearFocus();
            String url = mEtUrlInput.getText().toString();
            // load website if connected
            if (DownloadUtils.isConnected()) {
                mWebView.loadUrl(url);
            } else {
                CommonUtils.snackbar(v, getContext(), R.string.no_internet);
            }
        }
    }
}

File: app/src/main/java/faith/changliu/androiddemo/fragments/FragB.kt
package faith.changliu.androiddemo.fragments

import android.content.Context
import android.net.ConnectivityManager
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.webkit.WebChromeClient
import android.webkit.WebView
import android.webkit.WebViewClient
import faith.changliu.androiddemo.R
import faith.changliu.androiddemo.helpers.CommonUtils
import faith.changliu.androiddemo.helpers.DownloadUtils
import kotlinx.android.synthetic.main.frag_b.*

class FragB : Fragment() {

	companion object {
		val instance: FragB by lazy { FragB() }

	}

	private val cm: ConnectivityManager by lazy {
		context?.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
	}

	override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
		return inflater.inflate(R.layout.frag_b, container, false)
	}

	override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
		super.onViewCreated(view, savedInstanceState)

		// allow webView to load websites using javascript
		webView.settings.javaScriptEnabled = true

		// keep track of the loading process
		webView.webChromeClient = object : WebChromeClient() {
			override fun onProgressChanged(view: WebView?, newProgress: Int) {
				super.onProgressChanged(view, newProgress)

				if (newProgress in 1..99) {
					progressBar.progress = newProgress
					progressBar.visibility = View.VISIBLE
				} else {
					progressBar.visibility = View.GONE
				}

			}
		}

		// keep website loading in app
		webView.webViewClient = object : WebViewClient() {}

		// load input url
		mBtnLoadUrl.setOnClickListener {v ->
			mEtUrlInputLayout.clearFocus()
			val url = mEtUrlInput.text.toString()
			// load website if connected
			if (DownloadUtils.isConnected(cm)) {
				webView.loadUrl(url)
			} else {
				CommonUtils.snackbar(v, "No internet")
			}
		}
	}
}